site/en/docs/android/custom-tabs/guide-warmup-prefetch/index.md

```diff
@@ -0,0 +1,106 @@
+
+The basic steps are:
+
+1. Check via [CustomTabsClient](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient).[getPackageName()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#getPackageName(android.content.Context,java.util.List%3Cjava.lang.String%3E)) If the default browser supports Custom Tabs. If yes, bind to the CustomTabsService via [CustomTabsClient.bindCustomTabsService()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#bindCustomTabsService(android.content.Context,java.lang.String,androidx.browser.customtabs.CustomTabsServiceConnection)).
+2. Once connected to the CustomTabsService, in the [CustomTabsServiceConnection.onCustomTabsServiceConnected()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsServiceConnection#onCustomTabsServiceConnected(android.content.ComponentName,androidx.browser.customtabs.CustomTabsClient)) callback, do:
+    1. Warmup the browser process via [CustomTabsClient.warmup()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#warmup(long)).
+    2. Create a new [CustomTabsSession](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession) via [CustomTabsClient.newSession()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#newSession(androidx.browser.customtabs.CustomTabsCallback,int)).
+3. Optionally, prefetch web pages the user is likely to visit via [CustomTabsSession.mayLaunchUrl](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,android.os.Bundle,java.util.List%3Candroid.os.Bundle%3E))().
+4. When launching a new Custom Tab, pass the [CustomTabsSession](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession) to the [CustomTabsIntent.Builder](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsIntent.Builder) via the constructor `new CustomTabsIntent.Builder(session)`.
+
+Important: when targeting API level 30, [CustomTabsClient](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient).[getPackageName()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#getPackageName(android.content.Context,java.util.List%3Cjava.lang.String%3E)) requires you to add a queries section to your Android Manifest, declaring an intent-filter that matches browsers with Custom Tabs support.
+
+```xml
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+  …
+   <queries>
+        <intent>
+            <action android:name="android.support.customtabs.action.CustomTabsService" />
+        </intent>
+    </queries>
+</manifest>
+```
+
+Here is the full code:
```

Comment:
nit: this sentence is not needed. i'd remove it.

Reply:
I like to have something that separates the two snippets. rephrased